deleted input continue to fill the server log of errors

2015-02-27T16:56:27.538+01:00 ERROR [ExtractorFilter] Could not load extractors into cache. Returning empty list.
java.util.concurrent.ExecutionException: org.graylog2.database.NotFoundException: Input <54f08472e4b0e9a8aafe9256> not found!
        at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:299)
        at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:286)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116)
        at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:137)
        at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2348)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2320)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
        at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4739)
        at org.graylog2.filters.ExtractorFilter.loadExtractors(ExtractorFilter.java:73)
        at org.graylog2.filters.ExtractorFilter.filter(ExtractorFilter.java:59)
        at org.graylog2.buffers.processors.ServerProcessBufferProcessor.handleMessage(ServerProcessBufferProcessor.java:97)
        at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:71)
        at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:33)
        at com.lmax.disruptor.WorkProcessor.run(WorkProcessor.java:138)
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.graylog2.database.NotFoundException: Input <54f08472e4b0e9a8aafe9256> not found!
        at org.graylog2.inputs.InputServiceImpl.find(InputServiceImpl.java:115)
        at org.graylog2.filters.ExtractorFilter$1.call(ExtractorFilter.java:78)
        at org.graylog2.filters.ExtractorFilter$1.call(ExtractorFilter.java:73)
        at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4742)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
        ... 14 more
2015-02-27T16:56:27.539+01:00 ERROR [StaticFieldFilter] Could not load static fields into cache. Returning empty list.
java.util.concurrent.ExecutionException: org.graylog2.database.NotFoundException: Input <54f08472e4b0e9a8aafe9256> not found!
        at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:299)
        at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:286)
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116)
        at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:137)
        at com.google.common.cache.LocalCache$Segment.getAndRecordStats(LocalCache.java:2348)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2320)
        at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
        at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4739)
        at org.graylog2.filters.StaticFieldFilter.loadStaticFields(StaticFieldFilter.java:74)
        at org.graylog2.filters.StaticFieldFilter.filter(StaticFieldFilter.java:61)
        at org.graylog2.buffers.processors.ServerProcessBufferProcessor.handleMessage(ServerProcessBufferProcessor.java:97)
        at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:71)
        at org.graylog2.shared.buffers.processors.ProcessBufferProcessor.onEvent(ProcessBufferProcessor.java:33)
        at com.lmax.disruptor.WorkProcessor.run(WorkProcessor.java:138)
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
Caused by: org.graylog2.database.NotFoundException: Input <54f08472e4b0e9a8aafe9256> not found!
        at org.graylog2.inputs.InputServiceImpl.find(InputServiceImpl.java:115)
        at org.graylog2.filters.StaticFieldFilter$1.call(StaticFieldFilter.java:78)
        at org.graylog2.filters.StaticFieldFilter$1.call(StaticFieldFilter.java:74)
        at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4742)
        at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
        at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
        ... 14 more

--- a/graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java
+++ b/graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java
@@ -19,6 +19,7 @@ package org.graylog2.filters;
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
 import com.google.common.collect.Lists;
+import org.graylog2.database.NotFoundException;
 import org.graylog2.inputs.Input;
 import org.graylog2.inputs.InputService;
 import org.graylog2.plugin.Message;
@@ -75,17 +76,22 @@ public class ExtractorFilter implements MessageFilter {
                 public List<Extractor> call() throws Exception {
                     LOG.debug("Re-loading extractors for input <{}> into cache.", inputId);
 
-                    Input input = inputService.find(inputId);
+                    try {
+                        Input input = inputService.find(inputId);
 
-                    List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
+                        List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
 
-                    Collections.sort(sorted, new Comparator<Extractor>() {
-                        public int compare(Extractor e1, Extractor e2) {
-                            return e1.getOrder().intValue() - e2.getOrder().intValue();
-                        }
-                    });
+                        Collections.sort(sorted, new Comparator<Extractor>() {
+                            public int compare(Extractor e1, Extractor e2) {
+                                return e1.getOrder().intValue() - e2.getOrder().intValue();
+                            }
+                        });
 
-                    return sorted;
+                        return sorted;
+                    } catch (NotFoundException e) {
+                        LOG.warn("Unable to load input: {}", e.getMessage());
+                        return Collections.emptyList();
+                    }
                 }
             });
         } catch (ExecutionException e) {

--- a/graylog2-server/src/main/java/org/graylog2/filters/StaticFieldFilter.java
+++ b/graylog2-server/src/main/java/org/graylog2/filters/StaticFieldFilter.java
@@ -18,6 +18,7 @@ package org.graylog2.filters;
 
 import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
+import org.graylog2.database.NotFoundException;
 import org.graylog2.inputs.Input;
 import org.graylog2.inputs.InputService;
 import org.graylog2.plugin.Message;
@@ -75,8 +76,13 @@ public class StaticFieldFilter implements MessageFilter {
                 @Override
                 public List<Map.Entry<String, String>> call() throws Exception {
                     LOG.debug("Re-loading static fields for input <{}> into cache.", inputId);
-                    final Input input = inputService.find(inputId);
-                    return inputService.getStaticFields(input);
+                    try {
+                        final Input input = inputService.find(inputId);
+                        return inputService.getStaticFields(input);
+                    } catch (NotFoundException e) {
+                        LOG.warn("Unable to load input: {}", e.getMessage());
+                        return Collections.emptyList();
+                    }
                 }
             });
         } catch (ExecutionException e) {

